Лабораторная работа 5. Влад

LibrechtIsTheBest opened the pull request:
1) Не совсем понял про "Рутовые VC в этих NC одинаковые и имеют кнопку перехода на следующий экран (пуш на такой же VC)" и "Первый контроллер (первые два таба) содержит ...". Один это контроллер на сториборде или два одинаковых? 
2) Мне кажется, то как я сделал подсчет глубины контроллеров - это не очень хорошо.
3) Пока сделал все views с аутолейаутом, может позже сделаю авторесайз маски и/или задание координат вручную.

LibrechtIsTheBest committed:
Init commit.
All orientations supported, universal iPhone/iPad, Autolayout only.

ignatovSA commented:
1) Контроллер один и тот же. Дублирование сущностей это в любом случае плохо. Суть задания в том, чтобы на двух разных табах могла быть разная глубина стэка этих контроллеров, и всё это бы сохранялось при переключении табов.

LibrechtIsTheBest committed:
1) Number of view controller in stack is displaying in navigationItem
2) Remove NC subclasses
3) Replace AlertView with AlertController

LibrechtIsTheBest committed:
Move nivigationItem.title setting in viewDidLoad from viewDidAppear

LibrechtIsTheBest committed:
1) Add ScrollView to AuthorVC
2) Got rid of "count" property in ColorVC. Now navigationContrller.viewControllers.count is used
3) Add UIColor (HexColor) category for checking if hexString represent valid color
4) You can't enter invalid characters in colorTextField now, but if string valid it instantly change background color
5) weakAlert is used in alertController block

LibrechtIsTheBest committed:
Used UITextFieldDelegate protocol

LibrechtIsTheBest committed:
Code style

ignatovSA commented:
`@interface ColorVC : UIViewController <UITextFieldDelegate>`
Пользовалетям класса ColorVC совсем не обязательно знать, что он является делегатом `UITextField`. Лучше скрывать объявления о реализации протоколов в приватную часть, там где это не нужно.

ignatovSA commented:
`AboutVC->dismissView` - это презентуемый VC, и плохо, что он сам отвечает за своё пропадание с экрана. За этим должен следить тот VC, который его презентовал.

ignatovSA commented on LabWork5/UIColor+HexColor.m:
```diff
@@ -0,0 +1,65 @@
+        
+        *error = [NSError errorWithDomain:HexColorErrorDomain
+                                     code:HexColorErrorLenghtIsGreaterThan6
+                                 userInfo:@{NSLocalizedDescriptionKey: NSLocalizedString(@"Hex code contain illegal characters", nil)}];
```
Лучше придерживаться корпоративного стиля кодирования и не выравнивать длинные строки по двоеточию.

ignatovSA commented:
При отображении клавиатуры она закроет собой поле ввода.

LibrechtIsTheBest commented:
Исправил

LibrechtIsTheBest committed:
Hot fix

LibrechtIsTheBest committed:
Scroll Insets fix
